CMakeLists.txt

```diff
@@ -1,178 +1,152 @@
-
-###################################################################################################
-# - build options
+# build options
 
 option(BUILD_TESTS "Configure CMake to build tests" ON)
 option(BUILD_BENCHMARKS "Configure CMake to build (google) benchmarks" OFF)
-
 option(DISABLE_DEPRECATION_WARNING "Disable warnings generated from deprecated declarations." OFF)
-if(DISABLE_DEPRECATION_WARNING)
-  set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -Xcompiler -Wno-deprecated-declarations")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-deprecated-declarations")
-endif(DISABLE_DEPRECATION_WARNING)
+option(PER_THREAD_DEFAULT_STREAM "Build with per-thread default stream" OFF)
 
 # cudart can be statically linked or dynamically linked the python ecosystem wants dynamic linking
 option(CUDA_STATIC_RUNTIME "Statically link the CUDA runtime" OFF)
 
-option(PER_THREAD_DEFAULT_STREAM "Build with per-thread default stream" OFF)
-
-###################################################################################################
 # find packages we depend on
 
 find_package(CUDAToolkit REQUIRED)
 
-###################################################################################################
-# cmake modules
-
-set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake" ${CMAKE_MODULE_PATH})
-
-###################################################################################################
-# third-party dependencies
-
-include(RMM_thirdparty)
-
-###################################################################################################
-# per-thread default stream option
-# This needs to be defined first so tests and benchmarks can inherit it.
-
-if(PER_THREAD_DEFAULT_STREAM)
-    message(STATUS "Using per-thread default stream")
-    add_compile_definitions(CUDA_API_PER_THREAD_DEFAULT_STREAM)
-endif(PER_THREAD_DEFAULT_STREAM)
-
-###################################################################################################
 # library targets
 
 add_library(rmm INTERFACE)
 add_library(rmm::rmm ALIAS rmm)
 
-target_include_directories(rmm INTERFACE
-  "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
-  "$<INSTALL_INTERFACE:include>"
-  )
+target_include_directories(rmm INTERFACE "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+                                         "$<INSTALL_INTERFACE:include>")
 
 if(CUDA_STATIC_RUNTIME)
-  message(STATUS "Enabling static linking of cudart")
+  message(STATUS "RMM: Enabling static linking of cudart")
   target_link_libraries(rmm INTERFACE CUDA::cudart_static)
 else()
   target_link_libraries(rmm INTERFACE CUDA::cudart)
 endif(CUDA_STATIC_RUNTIME)
 
 target_link_libraries(rmm INTERFACE rmm::Thrust)
 target_link_libraries(rmm INTERFACE spdlog::spdlog_header_only)
+target_compile_features(rmm INTERFACE cxx_std_14 cuda_std_14)
+target_compile_options(rmm INTERFACE $<$<COMPILE_LANG_AND_ID:CXX,GNU,Clang>:-Werror
+                                     -Wno-error=deprecated-declarations>)
```

Comment:
Since this is an interface property, a downstream project will inherit it, making that entire project use `-Werror`, which might not be appreciated. I think this is an option that's better kept as a global option for the RMM project, which will prevent it from affecting downstream projects / applications.

Reply:
Will just move this to tests / benchmarks for now